packages/python-packages/apiview-stub-generator/apistub/_stub_generator.py

```diff
@@ -136,22 +136,34 @@ def install_extra_dependencies(self):
                 logging.info(f"Failed to install extra dependency: {extra}")
                 pass
 
-    def generate_tokens(self):
-        # Extract package to temp directory if it is wheel or sdist
-        if self.pkg_path.endswith(".whl") or self.pkg_path.endswith(".zip"):
-            pkg_root_path = self.wheel_path
-            pkg_name, version = self._parse_pkg_name()
-            namespace = self.get_module_root_name(pkg_root_path)
+    def _get_pkg_metadata(self):
+        metadata = get_metadata(self.pkg_path)
+        pkg_name = metadata.name
+        version = metadata.version
+        pkg_root_path = self.wheel_path or self.pkg_path
+        # TODO: pyproject.toml + whl/sdist: metadata.extras_require doesn't pick up optional dependencies
```

Comment:
Specifying optional dependencies in the pyproject.toml as per [[this](https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#dependencies-and-requirements)] doesn't work with whl/sdist. The extras are not included in the Provides-Extras section of the PKG-INFO/dist-info.

It _does_ work when the pkg path is the package itself.

Reply:
Currently, the extras require won't be an issue for anyone, since azure-quantum is the only package that needs extras and they're using setup.py.

Need to figure out how to get the optional deps picked up from the pyproject.toml.

Reply:
> It does work when the pkg path is the package itself.

Bit more clarity here? Are you saying that if you point this metadata parse at the distribution file specifically, it parses it properly?

Reply:
> Need to figure out how to get the optional deps picked up from the pyproject.toml.

I spent some time this morning digging a bit. I think this may be a part of how this gets encoded that we're not aware of. At least as far as `pkginfo` goes.

I say this because I did some local experimentation.

```python
>>> from pkginfo import Wheel
>>> mypackage = Wheel('C:/repo/azure-sdk-for-python/.artifacts/azure-core/azure_core-1.32.1-py3-none-any.whl')
>>> mypackage.requires_dist
['requests >=2.21.0', 'six >=1.11.0', 'typing-extensions >=4.6.0', "aiohttp >=3.0 ; extra == 'aio'"]
>>> mypackage.provides_extras
['aio']
```

You _can_ get the extras, you just need to do some filtering on the `requires_dist` from `pkginfo`.

Reply:
Ahh, I see! I tested again this morning and it worked, so this must've been the issue. Thank you @scbedd !